src/app.ts

```diff
@@ -36,7 +37,10 @@ export function generateServerlessRouter(fhirConfig: FhirConfig, supportedGeneri
             limit: '6mb',
         }),
     );
-
+    // Add cors handler before auth to allow pre-flight requests without auth.
+    // TODO update fhirConfig.server to supply cors options? Env variable offered as an alternative
+    const corsOptions: CorsOptions = process.env.CORS_OPTIONS ? JSON.parse(process.env.CORS_OPTIONS) : {};

```

Comment:
Could this be passed in as an optional parameter instead of a environment variable?

Reply:
Hi Robert,
Yes I agree, I made this change in our code base so I could pass in a config at deployment time.
The correct way to do it is for the interface module to provide a CorOptions structure in fhirConfig.server.  I was under the impression that you had done this, but if not I can take it on?  We need this change in place (with a new npm package) before we can make the router changes - what is the procedure for doing this?

Reply:
Yeah - I started to then decided not to. The `CorsOptions` object type comes from the **express** CORS middleware. We don't want any implementation detail leaking into the `interface`. This is to ensure that the `interface` can remain flexible and unopinionated on how the `interface` is implemented. 

An option for adding some sort of CORS option to the `interface` is to make it generic enough to work with any 'server implementation'. This requires a bit of research.

But if we add `CorsOption` as a parameter to the express router package (ie this package) we are able to provide the CORS ability to express router while keeping the interface free of any express details.